services/api/analyzers.js

```diff
@@ -265,16 +272,88 @@ function deleteAnalyzer(req, res, next) {
                 return next(createError(500, reply['error']['message']))
             }
             closeResponse()
-            models['analyzers'].findByIdAndRemove(id, (err) => {
-                if (err) {
-                    return next(createError(500, null, err))
-                }
-                res.status(204).send()
+
+            // get all corresponding events
+            models['events'].find({analyzerId: id})
+            .then((events) => {
+                // collect & delete all objects key of all events id
+                objKeys = []
+                forEach(events, (event) => {
+                    forEach(event.content, (value) => {
+                        // Each type of events has its own content structure. To
+                        // generalize, we assume object key is stored in string type.
+                        if (isString(value)) {
+                            objKeys.push(value);
+                        }
+                    });
+                });
+                // Delete all objects by keys.
+                return deleteObjects(objKeys).then(() => {return events});
+            })
+            .then((events) => {
+                // delete all corresponding events
+                const eventIds = map(events, event => event._id);
+                return models['events'].deleteMany({
+                    _id: {
+                        $in: eventIds,
+                    },
+                });
+            })
+            .then(() => {
+                // delete analyzer
+                return models['analyzers'].findByIdAndRemove(id);
+            })
+            .then(() => {
+                return res.status(204).send();
             })
+            .catch((err) => {
+                return next(createError(500, null, err))
+            });
+
         })
     })
 }
 
+function deleteObjects(keys) {
```

Comment:
It's better to factor out this function because not just deleting analyzers need to call it but deleting an event object will also need it.